cfgov/v1/jinja2tags.py

```diff
@@ -11,6 +13,27 @@ def date_formatter(dt, text_format=False):
     return extended_strftime(dt, format)
 
 
+def get_protected_url(context, page):
+    """Output a page's URL if it is live and '#' if not.
+
+    Used when inserting a link to a Wagtail page. If the page is live, we want
```

Comment:
I don't think this is used/ called at the time an editor is inserting a link -- it seems like it's scattered throughout templates and called wherever a developer remembered to wrap potential page links with this?  (Somewhat similar to the process we currently have for `parse_links`).  I know refactoring how it's called is out of scope for this PR, but don't quite follow this docstring.

Reply:
@richaagarwal you're right that this is very similar to the current `parse_links` setup. Including this template tag is dependent on the developer, and I'm not sure if it's applied across the board. Unfortunately there's no easy way to simplify this using a middleware like you're doing.

I would actually prefer to remove this logic completely but to reduce the scope of this PR I just tried to clean it up and add tests to clarify what's already there.

Regarding the docstring, would a better way to start the wording be

> This method can be used in templates to insert a link to this page that will be redacted if the page is not yet live. If the page is live...

Reply:
That sounds good -- thanks!